### ISSUE TITLE
EventDataProvider has a long delay

### ISSUE THREAD

**Issue by ryscheng** [hypercerts-org/oso#188]
### Which area(s) are affected? (leave empty if unsure)

Frontend

### To Reproduce

https://www.opensource.observer/artifactById/2222

### Describe the Bug

Goto the network panel of the developer tools.
You'll notice that it tries to query with `artifactIds=[]`. Then there's a 2 minute delay before it tries again with the real artifact IDs. Supabase isn't actually the slow thing....

### Expected Behavior

Should be fast.

**Comment by ryscheng** [comment]
Okay, the basic flow is:
1. Client => /artifact/namespace/name
2. Server middleware gets the Artifact from the database to get the `id`, redirects user to `/artifactById/[id]`
3. Client loads the generic artifactById page. Begins a client-side query for the same Artifact to get the metadata. At the same time the EventDataProvider begins a useless event query for empty [] artifacts. This gets cached by SWR.
4. Only after the metadata loads, does the title show up, passes in the `artifactId` to the EventDataProvider to begin a new query. however, because SWR cached the previous query, it waits until TTL expires before trying again

So! Lots of things to fix
1. SWR key should be unique per query params. Fixed here https://github.com/hypercerts-org/oso/pull/201
2. We should just skip any queries if artifactIds or eventTypes is empty
3. EventDataProvider should just get the `id` from the URL, rather than wait for the artifact metadata from the client-side query
4. [Nice to have] we should be able to pass the artifactMetadata from the initial server-side query to the client, so that client doesn't have to do the same exact query.
5. Debug why findById is so slow.

**Comment by ryscheng** [comment]
Fix 2 here
https://github.com/hypercerts-org/oso/pull/202

**Comment by ryscheng** [comment]
Fix 3 in Plasmic Project version 13.0.1

**Comment by ryscheng** [comment]
Fixed 4 in here
https://github.com/hypercerts-org/oso/pull/203

Also caching database fetches in this PR, so partially addresses 5

**Comment by ryscheng** [comment]
Just to re-cap the new flows

For `/artifactById/[id]`
1. Client gets a static page from the server
2. Client loads the generic artifactById page. Begins a client-side query for the artifact metadata. EventDataProvider also queries for data
3. Metadata (e.g. name) and event data is loaded asynchronously client-side

For `/artifact/[namespace]/[name]`
1. Client makes a request to the server
2. Server does the database query. This is cached via react.cache
3. The server passes the metadata into PlasmicComponent props directly so that metadata (e.g. name) loads faster.
4. EventDataProvider data fetching still happens client-side.

Should be the same for projects

**Comment by ryscheng** [comment]
Ah one, more thing, we need to change the Supabase queries to only query by date, so that we can cache better

**Comment by ryscheng** [comment]
Done here
https://github.com/hypercerts-org/oso/pull/204
